internal/internal_worker.go

```diff
@@ -978,8 +1082,11 @@ func (ae *activityExecutor) ExecuteWithActualArgs(ctx context.Context, args []in
 	result, resultErr := interceptor.ExecuteActivity(ctx, &ExecuteActivityInput{Args: args})
 	var serializedResult *commonpb.Payloads
 	if result != nil {
+		// Dynamic activities always return EncodedValues, skip encoding because result should already be encoded
```

Comment:
Didn't we say we wanted dynamic workflows and activities to be able to return any value?